person.rb

```diff
@@ -0,0 +1,65 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method #{__method__}"
+  end
+end
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    super()
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if @age >= 18 || @parent_permission == true
+
+    false
+  end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
+end
+
+class BaseDecorator < Nameable
```

Comment:
- Let's ensure you've added the getter/setter method for the nameable so it could be accessed. 
`attr_accessor :nameable`